src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/UserController.kt

```diff
@@ -0,0 +1,26 @@
+package org.vaccineimpact.api.app.controllers
+
+import org.vaccineimpact.api.app.ActionContext
+import org.vaccineimpact.api.app.controllers.endpoints.SecuredEndpoint
+import org.vaccineimpact.api.app.errors.UnknownObjectError
+import org.vaccineimpact.api.models.User
+
+class UserController(context: ControllerContext) : AbstractController(context)
+{
+    override val urlComponent = "/users"
+    override val endpoints = listOf(
+            SecuredEndpoint("/:username/", this::getUser, emptySet())
+    )
+
+    fun getUser(context: ActionContext): User
+    {
+        var userName = userName(context)
+        var user = repos.user().use { it.getUserByUsername(userName(context))  }
+        if (user == null)
+            throw UnknownObjectError(userName, "Username")
```

Comment:
I've been tending the follow the approach of making the repository endpoints return a non-nullable object, and then throwing the UnknownObjectException in the repository implementation. This means that if we use the same repository endpoint in multiple places we don't have to keep handling the null case.

`getUserByEmail` is a rather special case, because we want to throw a CredentialsException for that one.